feat: implement Amazee Lagoon provider integration, fixes #3188 (#5371)
ddev/ddev: docs/content/users/providers/lagoon.md

> # Amazee Lagoon Integration

Reviewer: ```suggestion
# Lagoon Integration
```

Lagoon exists as a stand-alone open source project and we want the integration to work for any Lagoon installation, so we can remove amazee.io as a call out.

> DDEV provides integration with the [Amazee Lagoon](https://lagoon.sh/), allowing Lagoon users to quickly download database and files from a Lagoon project to the local DDEV project.

Reviewer 2: Being picky, but rewording to avoid “download database” which is uncomfortable phrasing:

```suggestion
DDEV provides integration with the [Amazee Lagoon](https://lagoon.sh/), allowing Lagoon users to quickly download the files and database from a Lagoon project to the local DDEV project.
```

Author: You have never liked alliteration :)

Reviewer 2: I probably have but I can’t prove it!

> DDEV provides integration with the [Amazee Lagoon](https://lagoon.sh/), allowing Lagoon users to quickly download the files and database from a Lagoon project to the local DDEV project.

Reviewer: ```suggestion
DDEV provides integration with [Lagoon](https://lagoon.sh/), allowing users to quickly sync the files and database from a Lagoon environment to the local DDEV project.
```

> ## Amazee Lagoon Per-Project Configuration

Reviewer: ```suggestion
## Lagoon Per-Project Configuration
```

> 1. Check out the Lagoon project and configure it with [`ddev config`](../usage/commands.md#config). You’ll want to use [`ddev start`](../usage/commands.md#start) and make sure the basic functionality is working.

Reviewer 2: Run commands:

```suggestion
1. Check out the Lagoon project and configure it by running [`ddev config`](../usage/commands.md#config). You’ll want to run [`ddev start`](../usage/commands.md#start) and make sure the basic functionality is working.
```

2. Configure an SSH key at on the [Amazee Cloud Dashboard](https://dashboard.amazeeio.cloud/settings).

> 3. Add LAGOON_PROJECT and LAGOON_ENVIRONMENT variables to your project in 'web_environment' or a '.ddev/.env'. For example, `ddev config --web-environment-add="LAGOON_PROJECT=<project-name> LAGOON_ENVIRONMENT=<environment-name>"`.

Reviewer 2: Wrap environment variable names and filenames in backticks, clarify YAML vs. `.env` file, run commands:

```suggestion
3. Add `LAGOON_PROJECT` and `LAGOON_ENVIRONMENT` variables to your project using `'web_environment'` in its YAML configuration or a `.ddev/.env` file. For example, run `ddev config --web-environment-add="LAGOON_PROJECT=<project-name> LAGOON_ENVIRONMENT=<environment-name>"`.
```

Author: I wonder how this passed spellcheck.

> 4. `ddev auth ssh` if you haven't already done so. This will make your SSH key available in the web container of your project.

Reviewer 2: Run commands, edit for brevity. (No harm in running again, and someone will presumably know if they’ve run the command already—and maybe to run it again if their SSH key is not available in the web container.)

```suggestion
4. Run `ddev auth ssh` to make your SSH key available in the project’s web container.
```

> 2. Configure an SSH key at on the [Amazee Cloud Dashboard](https://dashboard.amazeeio.cloud/settings).
> 3. Add `LAGOON_PROJECT` and `LAGOON_ENVIRONMENT` variables to your project using `'web_environment'` in its YAML configuration or a `.ddev/.env` file. For example, run `ddev config --web-environment-add="LAGOON_PROJECT=<project-name> LAGOON_ENVIRONMENT=<environment-name>"`.
> 4. Run `ddev auth ssh` to make your SSH key available in the project’s web container.

Reviewer: Items 2 and 4 are directly related so I just moved them together. Also linking to Lagoon docs for SSH key instructions instead of amazee.io.

```suggestion
2. Add `LAGOON_PROJECT` and `LAGOON_ENVIRONMENT` variables to your project using `'web_environment'` in its YAML configuration or a `.ddev/.env` file. For example, run `ddev config --web-environment-add="LAGOON_PROJECT=<project-name> LAGOON_ENVIRONMENT=<environment-name>"`.
3. Configure an [SSH key](https://docs.lagoon.sh/using-lagoon-advanced/ssh/) for your Lagoon user.
4. Run `ddev auth ssh` to make your SSH key available in the project’s web container.
```

5. Run [`ddev restart`](../usage/commands.md#restart).
6. Run `ddev pull lagoon`. After you agree to the prompt, the current upstream databases and files will be downloaded.

> 7. Optionally use `ddev push lagoon` to push local files and database to Lagoon. The [`ddev push`](../usage/commands.md#push) command can potentially damage your production site, so we don’t recommend using it.

Reviewer 2: Run commands:

```suggestion
7. Optionally run `ddev push lagoon` to push local files and database to Lagoon. The [`ddev push`](../usage/commands.md#push) command can potentially damage your production site, so we don’t recommend using it.
```

## Usage

> * `ddev pull lagoon` will connect to Amazee Lagoon to download database and files. To skip downloading and importing either file or database assets, use the `--skip-files` and `--skip-db` flags.

Reviewer: ```suggestion
* `ddev pull lagoon` will connect to the Lagoon environment to download database and files. To skip downloading and importing either file or database assets, use the `--skip-files` or `--skip-db` flags.
```
